trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/chandler/FunctionHandler.java

```diff
@@ -661,7 +661,12 @@ private Result handleFunctionCallGivenNameAndArguments(final IDispatcher main, f
 			} else if (calleeProcInfo.getCType() != null) {
 				// we already know the parameters: do implicit casts and bool/int conversion
 				if (i >= calleeProcCType.getParameterTypes().length && calleeProcCType.hasVarArgs()) {
+					if (in.getCType() instanceof CPrimitive) {

```

Comment:
Shouldn't it be the underlying type here?

Reply:
I added a better check in 73c48b98b4d586f3fb469fd05d434677c9e0bb9a (not sure why it's not shown as outdated here...)

Reply:
Also I think it would be good, if `doIntegerPromotion` does not throw an exception for non-primitive types, but just returns the operand, but this should not be changed in this PR.